### ISSUE TITLE
Add City of Kingston scraper (PA-279)

### ISSUE THREAD

**Issue by oajirabot** [openaustralia/planningalerts#293]
<p><a href="http://www.kingston.vic.gov.au/web/map/?m=planning_register" class="external-link">http://www.kingston.vic.gov.au/web/map/?m=planning_register</a></p>

---

**Issue details imported from Jira:**

| Detail | Value |
| --- | --- |
| Assignee | Henare Degan |
| Reporter | Henare Degan |
| Created | 2011-02-19 12:46:23 |
| Updated | 2011-05-08 17:40:17 |

---
### Imported Jira Comments

**Henare Degan** - 2011-02-19 13:07:40

> <p>Last night on the train home I wrote a ScraperWiki scraper for Kingston as it was the most requested authority: <a href="http://scraperwiki.com/scrapers/city_of_kingston_development_applications/" class="external-link">http://scraperwiki.com/scrapers/city_of_kingston_development_applications/</a></p>
> 
> 
> <p>Because they seem to have all their DAs on that page, even going back to 2008, there's over 2300 records.</p>
> 
> 
> <p>After adding it to our parsers (<a href="https://github.com/henare/planningalerts-parsers/compare/PA-279" class="external-link">https://github.com/henare/planningalerts-parsers/compare/PA-279</a>) and running an import, we only get 100 records. This is because the ScraperWiki API by default just returns the 100 most recent items.</p>
> 
> 
> <p>My questions are:</p>
> 
> 
> <p>1) Do we want all 2300 records?</p>
> 
> 
> <p>I'd say yes - but this will presumably spam people that have already signed up for alerts with really old DAs the first time.</p>
> 
> 
> <p>2) If we do want all 2300, how should we go about this?</p>
> 
> 
> <p>I'm sure I've spoken with Matthew about the general issue of how we interact with ScraperWiki before but can't find any old conversations.</p>
> 
> 
> <p>As I understand it, ScraperWiki gives us the 100 most recently scraped DAs. Theoretically if there are over 100 new DAs in a scraper run, we'd miss any the next day when we parsed them because we didn't ask for them.</p>
> 
> 
> <p>We're also hitting the ScraperWiki API a number of times each run asking for the same thing because we do a few days of scraping for normal scrapers.</p>
> 
> 
> <p>When Matthew and I discussed it, we discussed switching to the API method "getDataByDate". Maybe this is all we need to do? (but we'd miss out on all the delicious old data Kingston has).</p>

**Henare Degan** - 2011-02-19 13:31:38

> <p>I've created a patch to use the ScraperWiki getDataByDate method instead: <a href="https://github.com/henare/planningalerts-parsers/commit/05dce0e87438ce6ecad7a2c928dfd994c05c8589" class="external-link">https://github.com/henare/planningalerts-parsers/commit/05dce0e87438ce6ecad7a2c928dfd994c05c8589</a></p>
> 
> 
> <p>To deploy that, we'd need to work out how we handle LGAs like the Blue Mountains that don't provide data_received. I think just set the date field in ScraperWiki to the date it was scraped. This would mean we'd correctly pick them up.</p>

**Henare Degan** - 2011-02-19 13:33:33

> <p>Assigning to Matthew to take a look.</p>
> 
> 
> <p>Matthew, please take a look at my proposal to change to getDataByDate (comment 2) and also I'd like your thoughts on importing all 2300-odd applications I've got from Kingston (comment 1).</p>
> 
> 
> <p>Thanks!</p>

**Matthew Landauer** - 2011-02-24 13:42:12

> <p>1. Before we can apply the getDataByDate we need to update all our scraperwiki scrapers to set the date. That's a special magic command isn't it?</p>
> 
> 
> <p>Perhaps we should split that task off into a separate ticket?</p>
> 
> 
> <p>2. As for importing all the old applications for Kingston, I think that probably would be unwise just right now because it could potentially generate email alerts with a thousand applications in it. There are no limits imposed on the number of alerts that can appear in an email. Also, as you said, the applications that would be in the alert would be very old.</p>
> 
> 
> <p>Maybe we should split the task of importing old alerts into a separate ticket so we can look at what would be the best way to do it there? It probably would only need to involve a small tweak to the way email alerts are sent out.</p>

**Henare Degan** - 2011-02-24 14:01:51

> <p>Thanks very much for taking a look at this Matthew.</p>
> 
> 
> <blockquote>
> <p>1. Before we can apply the getDataByDate we need to update all our scraperwiki scrapers to set the date. That's a special magic command isn't it?</p>
> 
> <p>Perhaps we should split that task off into a separate ticket?</p></blockquote>
> 
> 
> <p>Agreed: Created <a href="/browse/PA-282" title="Change ScraperWiki API method to getdatabydate"><del>PA-282</del></a>.</p>
> 
> 
> <blockquote>
> <p>2. As for importing all the old applications for Kingston, I think that probably would be unwise just right now because it could potentially generate email alerts with a thousand applications in it. There are no limits imposed on the number of alerts that can appear in an email. Also, as you said, the applications that would be in the alert would be very old.</p></blockquote>
> 
> 
> <p>I'm happy with that - created <a href="/browse/PA-283" title="Work out how to import old development applications"><del>PA-283</del></a> to track importing old DAs.</p>
> 
> 
> <p>I've put a dependency on <a href="/browse/PA-282" title="Change ScraperWiki API method to getdatabydate"><del>PA-282</del></a> for this ticket since if we deployed Kingston today, PA would just download the 100 most recently scraped DAs, which aren't necessarily the 100 most recent DAs.</p>

**Henare Degan** - 2011-05-08 17:40:17

> <p>Deployed.</p>
>
